clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -4108,17 +4126,24 @@ scheduler_tmpl_t<RecordType, ReaderType>::stop_speculation(output_ordinal_t outp
 }
 
 template <typename RecordType, typename ReaderType>
-void
+typename scheduler_tmpl_t<RecordType, ReaderType>::stream_status_t
 scheduler_tmpl_t<RecordType, ReaderType>::mark_input_eof(input_info_t &input)
 {
     assert(input.lock->owned_by_cur_thread());
     if (input.at_eof)
-        return;
+        return sched_type_t::STATUS_OK;
     input.at_eof = true;
     assert(live_input_count_.load(std::memory_order_acquire) > 0);
     live_input_count_.fetch_add(-1, std::memory_order_release);
```

Comment:
I know you didn't change this line in this PR, but I think the _safer_ way to do this update with assert is:

`int old_count = live_input_count_.fetch_add(-1, std::memory_order_release);`
`assert(old_count > 0);`

As-is, there's a brief gap when another thread could jump in and decrement the count between the assert and your decrement.

Reply:
Fixed.